lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -221,6 +240,32 @@ sub _get_first_batch {
     return ($helper->get_batches($project, 1), $project);
 }
 
+# _get_last_test_id is coupled with openqa-trigger-from-obs and tries to
+# extract id job returned by `job post` command as noted in openqa.cmd.log
+# returns empty string when log file doesn't exists or pattern didn't match
+# throws exception if OS error occurs
+sub _get_last_test_id {
+    my ($c, $alias) = @_;
+    my $helper = $c->obs_rsync;
+    my $home   = $helper->home;
+    # don't call get_first_batch: test info is not supported for batches yet
+    my $linkpath = Mojo::File->new($home, $alias, '.run_last');
+    my $folder;
+    $folder = readlink($linkpath);
+    my $cmdlog = Mojo::File->new($home, $alias, $folder)->child('openqa.cmd.log');
+    return '' unless -f $cmdlog;
+    my $fh = $cmdlog->open('<');
+    while (my $line = <$fh>) {
+        if ($line =~ /\{ id =\> ([1-9][0-9]*) \}/) {
+            close $fh;
+            return $1;
+        }
+    }
+    close $fh;
+    return '';
```

Comment:
please see my previous comment about Mojo::File. I think it should be possible to do something as simple as

```perl
return Mojo::File->new($home, $alias, '.run_last')->child('openqa.cmd.log')->slurp =~ /\{ id =\> ([1-9][0-9]*) \}/r // '';
```

Reply:
I don't think reading full file into memory is a good practice